crates/ltk_texture/src/tex/mod.rs

```diff
@@ -31,17 +34,89 @@ impl Tex {
     pub fn has_mipmaps(&self) -> bool {
         self.flags.contains(TextureFlags::HasMipMaps)
     }
+
+    /// Create a new Tex from an RGBA image with encoding options
+    ///
+    /// # Example
+    /// ```no_run
+    /// use ltk_texture::Tex;
+    /// use ltk_texture::tex::{EncodeOptions, Format, MipmapFilter};
+    /// use image::RgbaImage;
+    ///
+    /// let img = RgbaImage::new(256, 256);
+    ///
+    /// // Without mipmaps
+    /// let tex = Tex::from_rgba_image(&img, EncodeOptions::new(Format::Bc3)).unwrap();
+    ///
+    /// // With mipmaps
+    /// let tex_mips = Tex::from_rgba_image(
+    ///     &img,
+    ///     EncodeOptions::new(Format::Bc3).with_mipmaps()
+    /// ).unwrap();
+    ///
+    /// // With mipmaps and custom filter
+    /// let tex_lanczos = Tex::from_rgba_image(
+    ///     &img,
+    ///     EncodeOptions::new(Format::Bc3)
+    ///         .with_mipmaps()
+    ///         .with_mipmap_filter(MipmapFilter::Lanczos3)
+    /// ).unwrap();
+    /// ```
+    pub fn from_rgba_image(
```

Comment:
Incredibly nit but I don't like from_ methods being so computationally involved - could we have an encode method or something instead?

Reply:
I just mean the name of it, calling it encode instead of from_rgba_image - since from_rgba_image implies a simpler reshuffling of data (even though it does take encode params)